### DISCUSSION TITLE
should "query { 	_service { sdl 	} } " return @inaccessible directives in the sdl?

### DISCUSSION THREAD

**Discussion by manymotes** [apollographql/apollo-server#7319] [General] [answered]
Im trying to add inaccessible to my super graph. Im currently leveraging the _service{sdl} query to get the schema.graphql file. Sadly inaccessible directives are not being pulled back. Is this supported, if not, would make sense to have that directive in the response?

**Comment by trevor-scheer** [comment]
Are you using the `@apollo/subgraph` library? I would expect inaccessible directive usages to appear in the `_service { sdl }` query. If that's not the case, can you share a reproduction? (it might be worth posting this to the [federation repo](https://github.com/apollographql/federation) if you think there's an actual bug here)

**Comment by manymotes** [comment, marked as answer]
I was not linking fed 2 directives correctly. Sorry for the distraction.

**Reply by trevor-scheer** [reply]
No worries, thanks for following up. If you wouldn't mind sharing your solution in a bit more detail that might help the next person!

**Reply by manymotes** [reply]
Im using Netflix DGS, I assumed they were opting us into federation 2 by default. But I painfully found out we have to deliberately opt in by @linking like below in our schema:
```
extend schema
@link(url: "https://specs.apollo.dev/federation/v2.0", import: ["@inaccessible", "@key", "@extends", "@external"])
```
